pkg/operators/ustack/ustack_linux.go

```diff
@@ -47,6 +49,59 @@ func readUserStackMap(gadgetCtx operators.GadgetContext, userStackMap *ebpf.Map,
 		fmt.Fprintf(&addressesBuilder, "[%d]0x%016x; ", i, addr)
 	}
 	addressesStr := addressesBuilder.String()
+	buildIDStr := ""
 
-	return addressesStr, stackQueries, nil
+	// The buildIDMap is optional. Older gadgets won't have it.
+	if buildIDMap != nil && buildIDMap.MaxEntries() > 0 {
+		// struct bpf_stack_build_id is part of Linux UAPI:
+		// https://github.com/torvalds/linux/blob/v6.14/include/uapi/linux/bpf.h#L1451
+		type bpfStackBuildID struct {
+			status     int32
+			buildID    [unix.BPF_BUILD_ID_SIZE]uint8
+			offsetOrIP uint64 // Union of offset and ip
+		}
+		buildid := [ebpftypes.UserPerfMaxStackDepth]bpfStackBuildID{}
+		if buildIDMap.ValueSize() != uint32(unsafe.Sizeof(buildid)) {
+			logger.Warnf("build id map has unexpected value size %d instead of %d",
+				buildIDMap.ValueSize(), unsafe.Sizeof(buildid))
+			return "", "", nil, nil
+		}
```

Comment:
I think we can move this check to the init phase to avoid having to perform it on the hot path.